app/src/authorities/ActiveAlertAuthority.ts

```diff
@@ -0,0 +1,136 @@
+    ) {
+        this.factory = activeAlertsModelFactory;
+        void this.init(); // Hacky mchackface
+    }
+
+    public alertExists(world: number, zone: number): boolean {
+        return this._activeAlerts.has(this.mapKey(world, zone));
+    }
+
+    public getAlert(world: number, zone: number): ActiveAlertInterface {
+        const alert = this._activeAlerts.get(this.mapKey(world, zone));
+
+        if (alert) {
+            return alert;
+        }
+
+        throw new ApplicationException(`Unable to retrieve alert from active list! W: ${world} | Z: ${zone}`, 'ActiveAlertAuthority');
+    }
+
+    public async addAlert(mge: MetagameEventEvent): Promise<boolean> {
+        this._activeAlerts.set(this.mapKey(mge.world, mge.zone), {
+            alertId: alertId(mge),
+            instanceId: mge.instanceId,
+            world: mge.world,
+            zone: mge.zone,
+        });
+
+        // Commit to Database to persist between reboots
+        try {
+            const row = await this.factory.saveDocument({
+                alertId: alertId(mge),
+                instanceId: mge.instanceId,
+                world: mge.world,
+                zone: mge.zone,
+            });
+            ActiveAlertAuthority.logger.info(`Added ActiveAlert record for Alert #${row.alertId} | I:${row.instanceId} | W: ${mge.world} | Z:${mge.zone}`);
+
+            this.printActives();
+
+            return !!this.getAlert(mge.world, mge.zone);
+        } catch (err) {
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            throw new ApplicationException(`Unable to insert ActiveAlert into DB! ${err}`, 'ActiveAlertAuthority');
+        }
+    }
+
+    public async endAlert(mge: MetagameEventEvent): Promise<boolean> {
+        // Remove alert from in-memory list
+        this._activeAlerts.delete(this.mapKey(mge.world, mge.zone));
+
+        // Remove from database
+        try {
+            const res = await this.factory.model.deleteOne(
+                {alertId: alertId(mge)},
+            );
+
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            if (!res) {
+                ActiveAlertAuthority.logger.error(`No alerts were deleted on endAlert! ID: ${alertId(mge)}`);
+                return false;
+            }
+
+            return !this.getAlert(mge.world, mge.zone);
+        } catch (err) {
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            throw new ApplicationException(`Unable to insert alert into DB! ${err}`, 'ActiveAlertAuthority');
+        }
+
+    }
+
+    private async init(): Promise<boolean> {
+        ActiveAlertAuthority.logger.info('Initializing ActiveAlerts...');
+        // Pull the list out of the database and assign to in-memory array
+        let res: any[] = [];
+
+        try {
+            res = await this.factory.model.find();
```

Comment:
```suggestion
            res = await this.factory.model.find().exec();
```

Same as comment on #57 re `exec()` and `res`.